pkg/kubelet/prober/worker_test.go

```diff
@@ -530,6 +530,70 @@ func TestResultRunOnStartupCheckFailure(t *testing.T) {
 	}
 }
 
+func TestDoProbe_SidecarContainerWithRestartPolicy(t *testing.T) {
+	logger, ctx := ktesting.NewTestContext(t)
+	m := newTestManager()
+
+	// Test sidecar container (init container with restartPolicy=Always) behavior
+	w := newTestWorker(m, startup, v1.Probe{})
+
+	// Set container restart policy to Always (sidecar behavior)
+	restartPolicy := v1.ContainerRestartPolicyAlways
+	w.container.RestartPolicy = &restartPolicy
+
+	// Set pod restart policy to Never
+	w.pod.Spec.RestartPolicy = v1.RestartPolicyNever
+
+	// Create terminated status for sidecar container
+	terminatedStatus := getTestRunningStatus()
+	terminatedStatus.ContainerStatuses[0].State.Running = nil
+	terminatedStatus.ContainerStatuses[0].State.Terminated = &v1.ContainerStateTerminated{
+		StartedAt: metav1.Now(),
+	}
+
+	m.statusManager.SetPodStatus(logger, w.pod, terminatedStatus)
+
+	// Test: Terminated sidecar container with restartPolicy=Always should continue probing
+	// even when pod has restartPolicy=Never
+	if !w.doProbe(ctx) {
+		t.Error("Expected to continue probing for terminated sidecar container with restart policy Always")
+	}
+
+	// Verify result is set to Failure for terminated container
+	expectResult(t, w, results.Failure, "terminated sidecar with restart policy Always")
+}
+
+func TestDoProbe_RegularContainerBehaviorUnchanged(t *testing.T) {
```

Comment:
RegularContainerBehaviorUnchanged doesn't look valid in a general context (outside of this issue).
Can we be more specific?

TestDoProbe_TerminatedContainerWithRestartPolicyNever ?